File: transpiled/kotlin-sdk/src/main/kotlin/de/tutao/langApi/RuntimeInfo.kt
package de.tutao.langApi

import de.tutao.platformKit.appEnv.EnvType

class RuntimeInfo {
	companion object {
		val _isWorker: Boolean = true
		val _isNode: Boolean = true

		fun indexedDbIsSupported(): Boolean {
			return true
		}

		fun hasTouchEvent(): Boolean {
			return false
		}

		fun <E> globallyDefinedEnv(): EnvType? {
			return EnvType(null!!, null!!, null!!, null!!, null!!, null!!, null!!, null!!, null!!)
		}
	}
}

File: src/platform-kit/lang-api/common/runtimeInfo.ts
import { isNotNull } from "./functional"
import { Nullable } from "./types"

export class RuntimeInfo {
	public static readonly _isWorker: boolean = typeof WorkerGlobalScope !== "undefined" && self instanceof WorkerGlobalScope
	public static readonly _isNode: boolean =
		typeof process === "object" && typeof process.versions === "object" && typeof process.versions.node !== "undefined"

	public static indexedDbIsSupported(): boolean {
		try {
			return isNotNull(window.indexedDB)
		} catch (e) {
			return false
		}
	}

	public static hasTouchEvent(): boolean {
		return isNotNull(window.TouchEvent)
	}

	public static globallyDefinedEnv<E>(): Nullable<E> {
		return null as E
	}
}
